targets/TARGET_Maxim/TARGET_MAX32630/device/TOOLCHAIN_ARM_STD/MAX3263x.sct

```diff
@@ -11,19 +28,24 @@
 #  endif
 #endif
 
+; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
+#define VECTOR_SIZE 0x110
 #define Stack_Size MBED_CONF_TARGET_BOOT_STACK_SIZE
+#define RAM_FIXED_SIZE  (Stack_Size + VECTOR_SIZE)
 
-LR_IROM1 0x000000000 0x200000  {    ; load region size_region
-    ER_IROM1 0x000000000 0x200000  {  ; load address = execution address 
+LR_IROM1 MBED_APP_START MBED_APP_SIZE  {    ; load region size_region
+    ER_IROM1 MBED_APP_START MBED_APP_SIZE  {  ; load address = execution address 
         *.o (RESET, +First)
         *(InRoot$$Sections)
         .ANY (+RO)
     }
 
     ; [RAM] Vector table dynamic copy: 68 vectors * 4 bytes = 272 (0x110)
-    RW_IRAM1 (0x20000000+0x110) (0x80000-0x110-Stack_Size)  {  ; RW data
+    RW_IRAM1 (MBED_RAM_START+VECTOR_SIZE) (MBED_RAM_SIZE-RAM_FIXED_SIZE)  {  ; RW data
         .ANY (+RW +ZI)
     }
-    ARM_LIB_STACK (0x20000000+0x80000) EMPTY -Stack_Size { ; stack
+    ARM_LIB_HEAP AlignExpr(+0, +16) EMPTY (MBED_RAM_SIZE - RAM_FIXED_SIZE - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START)){ ; heap region growing up
```

Comment:
Thanks @hugueskamba - I've modified all the scatter files slightly so that the heap size is based on the following calculation: `MBED_RAM_SIZE - Stack_Size - (AlignExpr(ImageLimit(RW_IRAM1), 16) - MBED_RAM_START`. Does this make sense?

Reply:
It does not seem straightforward to me, I am not sure what other's opinions are.

What I suggested was `HEAP_SIZE = TOTAL_RAM_SIZE - REGIONS_IMMUTABLE_SIZES - ACTUAL_APP_SIZE`

where `TOTAL_RAM_SIZE` = `MBED_RAM_SIZE`
           `REGIONS_IMMUTABLE_SIZES` = `RAM_FIXED_SIZE`
           `ACTUAL_APP_SIZE` = `(APP_END_ADDRESS - APP_START_ADDRESS)`
           
where `APP_END_ADDRESS` = `AlignExpr(ImageLimit(RW_IRAM1), 16)`
           `APP_START_ADDRESS` = `(MBED_RAM_START + VECTOR_SIZE)`

Reply:
Okay - please see if my latest changes address your comments. 👌🏿

Reply:
Hey. I found that this calculation is incorrect and causes MBED to crash on MAX32XXX parts. Here is an open issue related to this: [#14712](https://github.com/ARMmbed/mbed-os/issues/14712)

My suggestion is for heap region calculation is:

`ARM_LIB_HEAP AlignExpr(+0, 16) EMPTY (MBED_RAM1_START + MBED_RAM_SIZE - RAM_FIXED_SIZE - AlignExpr(ImageLimit(RW_IRAM1), 16)){ ; heap region growing up
    }
`
which seems to resolve problem on MAX32630.

Reply:
@ahmetalincak Can you send a new pull request addressing this?